source/parsing/Preprocessor_macros.cpp

```diff
@@ -59,6 +59,17 @@ void Preprocessor::createBuiltInMacro(string_view name, int value, string_view v
 MacroActualArgumentListSyntax* Preprocessor::handleTopLevelMacro(Token directive) {
     auto macro = findMacro(directive);
     if (!macro.valid()) {
+        if (options.ignoreDirectives.find(directive.valueText().substr(1)) !=
+            options.ignoreDirectives.end()) {
+            SmallVectorSized<Trivia, 16> trivia;
+
+            trivia.append(createSimpleDirective(directive));
```

Comment:
If you follow the approach I outlined above then you don't need to (and really shouldn't) create a simple directive. The directive token will be taken by the MacroUsage syntax in the parent method.